tests/main_tests/test_data_check.py

```diff
@@ -0,0 +1,41 @@
+"""
+This script tests the data check function. This is a test.
+"""
+import pathlib
+
+import numpy as np
+import pandas as pd
+from pandas.testing import assert_frame_equal
+
+from src.utils.helper_functions.generate_test_data import data_generation
+
+
+def test_data_check():
+    """
+    check to ensure that the test data is generated correctly
+    """
+    file_out_path = "tests/data/test_data.csv"
+    assert pathlib.Path(file_out_path).exists()
+
+
+def test_data_check_identity():
+    """
+    check to ensure that the test data is generated correctly
+    """
+    file_out_path = "tests/data/test_data.csv"
+    num_of_rows = 100
+    df = data_generation(output_file_path=file_out_path, num_of_rows=num_of_rows)
+    assert len(df) == num_of_rows
+    np.random.seed(0)
```

Comment:
Consider providing some line-level comments here to describe what's being done.

Reply:
Good  point, I added more comments as to what is going on.